cpp/src/arima/arima_common.h

```diff
@@ -0,0 +1,210 @@
+/*
```

Comment:
Since this header file is needed by `arima.pyx`, this header file is to be considered as an "interface" header file. Thus needs to be moved under `cpp/include/arima/arima_common.h`. Since this file mostly contains a templated struct, I'd say it is just fine to move under `include/`.

Reply:
Ok. Btw, @teju85 should I move `batched_arima.hpp` and `batched_kalman.hpp` in the `include` folder too? The creation of the `include` folder happened while the original ARIMA PR was in review. And also is it ok to rename the file `arima.hpp` as we probably won't ever have a non-batched one?

Reply:
This is the rule of thumb I use to move headers into `include/`:
1. If a header file is being included from both `src/` or `src_prims/` or cython files.
2. Or if that header file is being included by another header which is already in `include/`.

Reply:
For now, let's be explicit about the naming. I'm ok with either one of the below choices:
1. keep it as `batched_arima.hpp`
2. rename it to `batched/arima.hpp` (and therefore will now have a lowest level `Batched::` namespace introduced)

Reply:
In this situation the headers are included only in Cython and their corresponding `.cu` files. So I just leave them where they are, right?

Reply:
Oh... then the point 1 of the rule-of-thumb applies to this case and thus we need to move both these headers into `include/`